ffi/diplomat/src/segmenter_line.rs

```diff
@@ -0,0 +1,50 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[diplomat::bridge]
+pub mod ffi {
+    use alloc::boxed::Box;
+    use core::convert::TryFrom;
+    use diplomat_runtime::DiplomatResult;
+    use icu_segmenter::LineBreakIterator;
+    use icu_segmenter::LineBreakSegmenter;
+
+    #[diplomat::opaque]
+    /// An ICU4X line-break segmenter, capable of finding breakpoints in strings.
+    /// See [the Rust docs](https://unicode-org.github.io/icu4x-docs/doc/icu_segmenter/struct.LineBreakSegmenter.html) for more information.
+    pub struct ICU4XLineBreakSegmenter(LineBreakSegmenter);
+
+    #[diplomat::opaque]
+    pub struct ICU4XLineBreakIterator<'a, 'b>(LineBreakIterator<'a, 'b, char>);
```

Comment:
I think this should be a single lifetime, covariance will mean that you won't be restricted by it, but it'll help the diplomat lifetime handling when we add it. 

There's really no benefit to maximum flexibility on lifetimes for types that are primarily used by environments without them.

Reply:
Done